boards/seeed/xiao_rp2040/board.cmake

```
@@ -0,0 +1,39 @@
+# SPDX-License-Identifier: Apache-2.0
+
+# This configuration allows selecting what debug adapter debugging to use
+# for the Xiao RP2040 by a command-line argument.
+# It is mainly intended to support both the 'picoprobe' and 'raspberrypi-swd'
+# adapter described in "Getting started with Raspberry Pi Pico".
+# And any other SWD debug adapter might also be usable with this configuration.
+
+# Note that the Xiao RP2040 SWD pins are test points on the underside
+# of the board.
+
+# Set RPI_PICO_DEBUG_ADAPTER to select debug adapter by command-line arguments.
+# e.g.) west build -b rpi_pico -- -DRPI_PICO_DEBUG_ADAPTER=raspberrypi-swd
+# The value is treated as a part of an interface file name that
+# the debugger's configuration file.
+# The value must be the 'stem' part of the name of one of the files
+# in the openocd interface configuration file.
+# The setting is store to CMakeCache.txt.
+if ("${RPI_PICO_DEBUG_ADAPTER}" STREQUAL "")
+	set(RPI_PICO_DEBUG_ADAPTER "cmsis-dap")
+endif()
```

minor nit, can be reduced to:
```suggestion
set_ifndef(RPI_PICO_DEBUG_ADAPTER "cmsis-dap")
```